tests/apollo/test_skvbc_commit_path.py

```diff
@@ -85,24 +113,58 @@ async def test_fast_to_slow_path(self, bft_network, tracker):
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
 
         # Crash C+1 replicas excluding the primary - this ensures that the slow path will be used
         # without forcing a view change
         crash_targets = random.sample(bft_network.all_replicas(without={0}), bft_network.config.c + 1)
         bft_network.stop_replicas(crash_targets)
 
-        await bft_network.wait_for_slow_path_to_be_prevalent(
-            run_ops=lambda: skvbc.run_concurrent_ops(num_ops=NUM_OPS, write_weight=1), threshold=NUM_OPS)
+        await bft_network.wait_for_consensus_path(path_type=ConsensusPathType.SLOW,
+                                                  run_ops=lambda: self.send_kvs_sequentially(skvbc, op_count),
+                                                  threshold=op_count)
 
     @with_trio
-    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n >= 6, rotate_keys=True)
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n >= 6 and c > 0, rotate_keys=ROTATE_KEYS)
+    @verify_linearizability()
+    async def test_fast_to_fast_with_threshold_path(self, bft_network: 'BftTestNetwork', tracker: 'SkvbcTracker'):
+        """
+        This test check the transitions from fast to fast_with_threshold commit path.
+        First we write a series of K/V entries making sure we stay on the fast path.
+
+        Once the first series of K/V writes have been executed we bring down a single replica,
+        which should trigger a transition to the fast_with_threshold path.
+
+        We send a new series of K/V writes and make sure they
+        have been executed using the fast_with_threshold commit path.
+        """
+
+        primary_num = 0
+        fast_ops = 5
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        ops_for_transition = int(self.EVALUATION_PERIOD_SEQUENCES * 1.1)
+
+        # Initially all replicas are running on the fast path
+        await bft_network.wait_for_consensus_path(path_type=ConsensusPathType.OPTIMISTIC_FAST,
+                                                  run_ops=lambda: self.send_kvs_sequentially(skvbc, fast_ops),
+                                                  threshold=fast_ops)
+
+        crash_targets = random.sample(bft_network.all_replicas(without={primary_num}), bft_network.config.c)
+        bft_network.stop_replicas(crash_targets)
+
+        await bft_network.wait_for_consensus_path(path_type=ConsensusPathType.FAST_WITH_THRESHOLD,
+                                                  run_ops=lambda: self.send_kvs_sequentially(skvbc, ops_for_transition),
+                                                  threshold=1)
+
+        slow_commits = await bft_network.commit_count_by_path(ConsensusPathType.SLOW)
+        assert slow_commits > 0, f"Invalid slow commit count on transition from fast path to fast with threshold, " \
+                                 f"slow count {slow_commits}"
```

Comment:
Is the purpose of the assert here to make sure that transitions between fast optimistic to fast with threshold involve commits on the slow path?

Reply:
Yes, I'll change the assertion's text